phoenix-core/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -1171,14 +1183,69 @@ public ColumnRef resolveColumn(String schemaName, String tableName, String colNa
-                                tableRef.getTable().getParentTableName().getString());
+                        TableRef parentTableRef = null;
+                        if (tableName == null) {
+                          // No table specified
+                          Iterator<TableRef> iterator = localIndexDataTables.iterator();
+                          while (iterator.hasNext()) {
+                              TableRef searchTableRef = iterator.next();
+                              try {
+                                  searchTableRef.getTable().getColumnForColumnName(colName);
+                                  if (parentTableRef != null) {
+                                      throw new AmbiguousColumnException(colName);
+                                  }
+                                  parentTableRef = searchTableRef;
+                              } catch (ColumnNotFoundException e2) {
+                              }
+                          }
+                          if (parentTableRef == null) {
+                              throw new ColumnNotFoundException(schemaName, tableName, null, colName);
+                          }
+                      } else {
+                        // table specified
+                        parentTableRef = resolveLocalIndexDataTable( schemaName, tableName);
+                      }
                         colRef = new ColumnRef(parentTableRef,
                                 IndexUtil.getDataColumnFamilyName(colName),
                                 IndexUtil.getDataColumnName(colName));
                     } catch (TableNotFoundException te) {
-                        throw e;
+                        //columfamily format
+                        TableRef theTableRef = null;
+                        PColumnFamily theColumnFamily = null;
+                        PColumn theColumn = null;
+                        if (schemaName != null) {
+                            try {
+                                // Try schemaName as the tableName and use tableName as column family name
+                                theTableRef = resolveLocalIndexDataTable(null, schemaName);
+                                theColumnFamily = theTableRef.getTable().getColumnFamily(tableName);
+                                theColumn = theColumnFamily.getPColumnForColumnName(colName);
+                            } catch (MetaDataEntityNotFoundException e2) {
+                            }
+                        }
+                        if (theColumn == null) {
+                            // Try using the tableName as a columnFamily reference instead
+                            // and resolve column in each column family.
+                            Iterator<TableRef> iterator = localIndexDataTables.iterator();
+                            while (iterator.hasNext()) {
+                                TableRef searchTableRef = iterator.next();
```

Comment:
Same as above, could be replaced with `for (TableRef localIndexDataTableRef : localIndexDataTables)`

Reply:
Again, this copies the similar code from the parent class, and I wanted to keep that obvious.
We could change it in both places, but that would add even more noise to the patch.